### DISCUSSION TITLE
Should we use Java 11 APIs for unit testing?

### DISCUSSION THREAD

**Discussion by jkwatson** [open-telemetry/opentelemetry-java#3037] [Q&A]
We already require java 11+ for builds. Could we start using it in unit/integration tests?

Advantages:
* `var`
* `Map.of(...)`, `List.of(...)` etc
* HttpClient for integration testing

Disadvantages:
* Need to make sure that we keep java 8 only for all production artifacts.

**Comment by Oberon00** [comment]
The second point is covered by Guava. For the third point: How about using Java 11 only for integration tests?

I think, but am not sure, there is a gradle configuration that runs the unit tests with a Java 8 JVM. These would be nice to keep.

**Reply by anuraaga** [reply]
We require Java 11 for builds but still run tests on Java 8 (invoked from Java 11). We wouldn't be able to do so if we started using Java 11 features for the tests though. So I'd say stick to Java 8 since running tests on the versions we support is important I guess.
